### ISSUE TITLE
upgrade to 2022.06 fails with tomcat exeption "java.lang.IllegalStateException: Error starting child"

### ISSUE THREAD

**Issue by StefanSa** [uyuni-project/uyuni#5607]
# Problem description

Hi there,
last upgrade 2022.06 from uyuni server fails. Tomcat server start / fails with an exeption.
`java.lang.IllegalStateException: Error starting child`

## Version of Uyuni Server and Proxy (if used)

`zypper info Uyuni-Server-release`
```
Repository-Daten werden geladen...
Installierte Pakete werden gelesen...


Informationen zu Paket Uyuni-Server-release:
--------------------------------------------
Repository         : Uyuni Server Stable
Name               : Uyuni-Server-release
Version            : 2022.06-183.9.uyuni2
Arch               : x86_64
Anbieter           : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level      : Stufe 3
Installierte Größe : 1,4 KiB
Installiert        : Ja
Status             : veraltet (Version 2022.03-172.4.uyuni1 installiert)
Quellpaket         : Uyuni-Server-release-2022.06-183.9.uyuni2.src
Zusammenfassung    : Uyuni Server
Beschreibung       :
    Uyuni lets you efficiently manage physical, virtual,
    and cloud-based Linux systems. It provides automated and cost-effective
    configuration and software management, asset management, and system
    provisioning.
```

## Details about the issue
```
27-Jun-2022 09:26:42.988 SCHWERWIEGEND [main] org.apache.catalina.startup.HostConfig.deployDirectory Error deploying web application directory [/srv/tomcat/webapps/rhn]
        java.lang.IllegalStateException: Error starting child
                at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:720)
                at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:690)
                at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
                at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1132)
                at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1865)
                at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
                at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
                at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
                at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118)
                at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1044)
                at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:429)
                at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1575)
                at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
                at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
                at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
                at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
                at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
                at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
                at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
                at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
                at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
                at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140)
                at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
                at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:566)
                at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
                at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)
        Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/rhn]]
                at org.apache.catalina.util.LifecycleBase.handleSubClassException(LifecycleBase.java:440)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:198)
                at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
                ... 37 more
        Caused by: java.lang.NullPointerException
                at org.apache.tomcat.util.scan.StandardJarScanner.process(StandardJarScanner.java:382)
                at org.apache.tomcat.util.scan.StandardJarScanner.scan(StandardJarScanner.java:195)
                at org.apache.catalina.startup.ContextConfig.processJarsForWebFragments(ContextConfig.java:1971)
                at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1129)
                at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:775)
                at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:301)
                at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
                at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5044)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                ... 38 more
```

**Comment by mcalmer** [comment]
Are you migrated to Leap 15.4? AFAIK this exists only on 15.4.

Check if the latest version of postgresql-jdbc , ongres-scram and ongres-scram-client is installed.
They should require also ongres-strinprep* packages.

**Comment by StefanSa** [comment]
Quick Info,
a test with the last backup and the server major upgrade (server-migrator.sh), ran without problems. 
We have successfully migrated to v15.4 // 2022.06.

**Comment by StefanSa** [comment]
Quick Info,
a test with the last backup and the server major upgrade (server-migrator.sh), ran without problems. 
We have successfully migrated to v15.4 // 2022.06.

**Comment by mbussolotto** [comment]
closing the issue since @StefanSa it seems fix ( 2022.06 it's a major upgrade, so it requires `server-migrator.sh` in order to upgrade the OS version as well).

**Comment by SSchott** [comment]
For the record, I had the very same issue. I was going with a minor update within 15.3, and I lost access to the GUI. To me it seems as if one of the packages got updated for 15.4, even though I did not request that. I am right now seeing if, as OP, doing a major update fixes the issue.
